runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -202,36 +230,53 @@ void registerTable(@Nonnull String namespace,
         while (numRetries-- > 0) {
             // Schema validation to ensure that there is either proper modification of the schema across open calls.
             // Or no modification to the protobuf files.
+            /**
+             * Caller is opening a new table with a map of protobufFilename -> protobufFileDescriptor pairs
+             * Here are the cases that can happen:
+             * 1. No entry in TableRegistry -> New table
```

Comment:
If a table is new(never seen before), how(between what) can we detect a mismatch?

Reply:
We can still detect a mismatch if a new unseen table tries to open and existing protobuf file which has now changed.
For example if someone change the corfu_store_metadata.proto and opened a new table, we can detect that there is a schema change with this PR. Before it was not possible.